website/server/libs/cron.js

```diff
@@ -66,20 +66,37 @@ function grantEndOfTheMonthPerks (user, now) {
 
   if (elapsedMonths > 0) {
     plan.dateUpdated = now;
-    // For every month, inc their "consecutive months" counter. Give perks based on consecutive blocks
-    // If they already got perks for those blocks (eg, 6mo subscription, subscription gifts, etc) - then dec the offset until it hits 0
-    _.defaults(plan.consecutive, {count: 0, offset: 0, trinkets: 0, gemCapExtra: 0});
-
-    for (let i = 0; i < elapsedMonths; i++) {
-      plan.consecutive.count++;
-
-      if (plan.consecutive.offset > 1) {
-        plan.consecutive.offset--;
-      } else if (plan.consecutive.count % 3 === 0) { // every 3 months
-        if (plan.consecutive.offset === 1) plan.consecutive.offset--;
-        plan.consecutive.trinkets++;
-        plan.consecutive.gemCapExtra += 5;
-        if (plan.consecutive.gemCapExtra > 25) plan.consecutive.gemCapExtra = 25; // cap it at 50 (hard 25 limit + extra 25)
+
+    let dateCreatedMoment = moment(plan.dateCreated).startOf('month');
```

Comment:
As far as I'm aware, there's no issue with it being updated when a new subscription is started, but that hasn't been done for all the existing subscriptions, so any code that uses dateCreated to calculate consecutive-month benefits will be inaccurate for many existing subscriptions and in a way that's less predictable than the current bug and therefore harder for us to do individual fixes for. I'm not saying we can't use this method - if it works for new subscriptions that will be great - but we might need to have some way of marking old subscriptions where we know the dateCreated is inaccurate to help us do manual fixes. E.g., we could do a binary flag that's set to true on all relevant subscriptions before this PR goes live and then is never set to true on any new subscriptions.